src/sacn/dmx_merger.c

```diff
@@ -0,0 +1,314 @@
+{
+  return NULL;
+}
+
+/*!
+ * \brief Updates the source data and recalculate outputs.
+ *
+ * The direct method to change source data.  This causes the merger to recalculate the outputs.
+ * If you are processing sACN packets, you may prefer dmx_merger_update_source_from_sacn().
+ *
+ * \param[in] merger The handle to the merger.
+ * \param[in] source The id of the source to modify.
+ * \param[in] new_values The new DMX values to be copied in. This may be NULL if the source is only updating the
+ * priority or address_priorities.
+ * \param[in] new_values_count The length of new_values. May be 0 if the source is only updating the priority or
+ * address_priorities.
+ * \param[in] priority The universe-level priority of the source.
+ * \param[in] address_priorities The per-address priority values to be copied in.  This may be NULL if the source is not
+ * sending per-address priorities, or is only updating other parameters.
+ * \param[in] address_priorities_count The length of address_priorities.  May be 0 if the source is not sending these
+ * priorities, or is only updating other parameters.
+ * \return #kEtcPalErrOk: Source updated and merge completed.
+ * \return #kEtcPalErrInvalid: Invalid parameter provided.
+ * \return #kEtcPalErrNotInit: Module not initialized.
+ * \return #kEtcPalErrNotFound: Handle does not correspond to a valid source or merger.
+ * \return #kEtcPalErrSys: An internal library or system call error occurred.
+ */
+etcpal_error_t sacn_dmx_merger_update_source_data(sacn_dmx_merger_t merger, source_id_t source,
+                                                  const uint8_t* new_values, size_t new_values_count, uint8_t priority,
+                                                  const uint8_t* address_priorities, size_t address_priorities_count)
+{
+  return kEtcPalErrNotImpl;
+}
+
+/*!
+ * \brief Updates the source data from a sACN packet and recalculate outputs.
+ *
+ * Processes data passed from the sACN receiver's SacnUniverseDataCallback handler.  This causes the merger to
+ * recalculate the outputs.
+ *
+ * \param[in] merger The handle to the merger.
+ * \param[in] source The id of the source to modify.
+ * \param[in] header The sACN header.  Must NOT be NULL.
+ * \param[in] pdata The sACN data.
+ * \return #kEtcPalErrOk: Source updated and merge completed.
+ * \return #kEtcPalErrInvalid: Invalid parameter provided.
+ * \return #kEtcPalErrNotInit: Module not initialized.
+ * \return #kEtcPalErrNotFound: Handle does not correspond to a valid source or merger.
+ * \return #kEtcPalErrSys: An internal library or system call error occurred.
+ */
+// TODO: If Receiver API changes to notify both values and per-address priority data in the same callback, this should
+// change!!
+etcpal_error_t sacn_dmx_merger_update_source_from_sacn(sacn_dmx_merger_t merger, source_id_t source,
+                                                       const SacnHeaderData* header, const uint8_t* pdata)
```

Comment:
If I'm calling this from SacnUniverseDataCallback then I'm going to have to track the source_id_t since the callback will only give me the CID. Unless if source_id_t could become a part of the receiver API as well?

Reply:
That was an oversight, Good Catch!  I've added a get_id function for general use, and removed the id from the parameters of this function.